crates/miden-block-prover/src/local_block_prover.rs

```diff
@@ -0,0 +1,265 @@
+        // taken into account?
+        let (
+            _batches,
+            mut account_updated_witnesses,
+            output_note_batches,
+            block_note_tree,
+            created_nullifiers,
+            chain_mmr,
+            prev_block_header,
+        ) = proposed_block.into_parts();
+
+        let prev_block_commitment = prev_block_header.hash();
+
+        // Get the root of the block note tree.
+        // --------------------------------------------------------------------------------------------
+
+        // TODO: Do we need the full tree in proposed block or would the root be sufficient? We
+        // can reconstruct the tree from the output note batches, so the question is whether it
+        // should be included in the proven block to be readily accessible or if recomputing it is
+        // fine.
+        let note_root = block_note_tree.root();
+
+        // Insert the created nullifiers into the nullifier tree to compute its new root.
+        // --------------------------------------------------------------------------------------------
+
+        let (created_nullifiers, new_nullifier_root) =
+            compute_nullifiers(created_nullifiers, &prev_block_header, block_num)?;
+
+        // Insert the previous block header into the block chain MMR to get the new chain root.
+        // --------------------------------------------------------------------------------------------
+
+        let new_chain_root = compute_chain_root(chain_mmr, prev_block_header);
+
+        // Insert the state commitments of updated accounts into the account tree to compute its new
+        // root.
+        // --------------------------------------------------------------------------------------------
+
+        let new_account_root = compute_account_root(&mut account_updated_witnesses)?;
+
+        // Transform the account update witnesses into block account updates.
+        // --------------------------------------------------------------------------------------------
+
+        let updated_accounts = account_updated_witnesses
+            .into_iter()
+            .map(|(account_id, update_witness)| {
+                let (
+                    _initial_state_commitment,
+                    final_state_commitment,
+                    // Note that compute_account_root took out this value so it should not be used.
+                    _initial_state_proof,
+                    details,
+                    transactions,
+                ) = update_witness.into_parts();
+                BlockAccountUpdate::new(account_id, final_state_commitment, details, transactions)
+            })
+            .collect();
+
+        // Construct the new block header.
+        // --------------------------------------------------------------------------------------------
+
+        // TODO: Where is this defined? Should we rename this to `protocol_version`, if it is that?
+        let version = 0;
```

Comment:
We haven't used it anywhere yet and I actually haven't thought through how this could be used. For example, this could be protocol version or block version (e.g., if we change serialization format of the block) or both. Let's create an issue to come back to this.

Reply:
Makes sense. Created https://github.com/0xPolygonMiden/miden-base/issues/1155.